include/rsp.inc

```diff
@@ -473,138 +732,229 @@ makeOpInstruction vsubc, 0b010101
+    #   op vt, offset, base
+    #   op vt.e, offset, base
+    .macro \name vt, elementOrOffset, offsetOrBase, baseOrEmpty
+        \elMode\()VectorElements
+
+        .ifnb \baseOrEmpty
+            lsInstruction \mode, \opcode, \vt, \elementOrOffset, \offsetOrBase, \baseOrEmpty, \size, \rangemin, \rangemax
+            .exitm
+        .endif
+
+        \elMode\()VectorAccessors
+        encodeVectorRegs
+
+        .iflt (\vt)
+            .error "Invalid element"
+            .exitm
+        .endif
+
+        # Dot-style syntax is encoded as numbers >= 0x200
+        .if (\vt >= 0x200)
+            regVectorAccessors
+            lsInstruction \mode, \opcode, reg.\vt, (\vt & 0xF), \elementOrOffset, \offsetOrBase, \size, \rangemin, \rangemax
+        .else
+            lsInstruction \mode, \opcode, \vt, 0, \elementOrOffset, \offsetOrBase, \size, \rangemin, \rangemax
+        .endif
+    .endm
+.endm
+
+.macro makeLsInstructionQuad mode, elMode, name, opcode
+    makeLsInstruction \mode, \elMode, \name, \opcode, 16, -1024, 1008
+.endm
+
+.macro makeLsInstructionDouble mode, elMode, name, opcode
+    makeLsInstruction \mode, \elMode, \name, \opcode, 8, -512, 504
+.endm
+
+.macro makeLsInstructionLong mode, elMode, name, opcode
+    makeLsInstruction \mode, \elMode, \name, \opcode, 4, -256, 252
+.endm
+
+.macro makeLsInstructionShort mode, elMode, name, opcode
+    makeLsInstruction \mode, \elMode, \name, \opcode, 2, -128, 126
+.endm
+
+.macro makeLsInstructionByte mode, elMode, name, opcode
+    makeLsInstruction \mode, \elMode, \name, \opcode, 1, -64, 63
+.endm
+
 /** @brief Load Byte into Vector Register */
-makeLsInstructionByte load, lbv, 0b00000
+makeLsInstructionByte load, byte, lbv, 0b00000
 /** @brief Load Double into Vector Register */
-makeLsInstructionDouble load, ldv, 0b00011
+makeLsInstructionDouble load, byte, ldv, 0b00011
 /** @brief Load Packed Fourth into Vector Register */
-makeLsInstructionQuad load, lfv, 0b01001
+makeLsInstructionQuad load, lane, lfv, 0b01001
 /** @brief Load Packed Half into Vector Register */
-makeLsInstructionQuad load, lhv, 0b01000
+makeLsInstructionQuad load, lane, lhv, 0b01000
 /** @brief Load Long into Vector Register */
-makeLsInstructionLong load, llv, 0b00010
+makeLsInstructionLong load, byte, llv, 0b00010
 /** @brief Load Packed Bytes into Vector Register */
-makeLsInstructionDouble load, lpv, 0b00110
+makeLsInstructionDouble load, lane, lpv, 0b00110
 /** @brief Load Quad into Vector Register */
-makeLsInstructionQuad load, lqv, 0b00100
+makeLsInstructionQuad load, byte, lqv, 0b00100
 /** @brief Load Quad (Rest) into Vector Register */
-makeLsInstructionQuad load, lrv, 0b00101
+makeLsInstructionQuad load, byte, lrv, 0b00101
 /** @brief Load Short into Vector Register */
-makeLsInstructionShort load, lsv, 0b00001
+makeLsInstructionShort load, byte, lsv, 0b00001
 /** @brief Load Transpose into Vector Register */
-makeLsInstructionQuad load, ltv, 0b01011
+makeLsInstructionQuad load, byte, ltv, 0b01011
```

Comment:
ltv and stv are an exception :)

In theory the element is treated as a byte offset, but in practice the lowest bit is ignored. So semantically speaking I would say that it should only accept a lane, but encoding-wise a byte offset should be written.

Reply:
I've added a special encoding for ltv and stv